lib/netfilter/queue_msg_obj.c

```diff
@@ -402,15 +402,18 @@ const uint8_t *nfnl_queue_msg_get_hwaddr(const struct nfnl_queue_msg *msg,
 	return msg->queue_msg_hwaddr;
 }
 
-int nfnl_queue_msg_set_payload(struct nfnl_queue_msg *msg, uint8_t *payload,
-			       int len)
+int nfnl_queue_msg_set_payload(struct nfnl_queue_msg *msg, const void *payload,
+			       unsigned int len)
 {
-	free(msg->queue_msg_payload);
```

Comment:
Forgot to add (I can split to two commits if you want) This fix case:
1. message has already some payload
2. malloc() failed

in that case `msg->queue_msg_payload_len` stay non-zero, `msg->queue_msg_payload`  become NULL.

It is fixed with my patch